.github/workflows/build-container.yaml

```diff
@@ -54,38 +49,42 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
       - id: meta
-        if: ${{ env.PUBLISH }}
         name: Extract metadata (tags, labels) for Docker
         uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 #v5.7.0
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=ref,event=tag
+            type=ref,event=pr
+            type=ref,event=branch
       - id: name-artifact
         env:
           REPO: ${{ github.repository }}
-          RELEASE: ${{ github.event_name == 'release' }}
-          TAG: ${{ steps.meta.outputs.tags || inputs.tag || env.TAG }}
+          TAG: ${{ steps.meta.outputs.tags }}
+          LABEL: ${{ steps.meta.outputs.labels }}
         run: |
-          if [ -z "$RELEASE" ]; then
-            echo "name=tag-release" >> "$GITHUB_OUTPUT"
-          else
-            tag=$(echo $TAG | sed -e "s+.*\:++")
-            echo "name=tag-$tag" >> "$GITHUB_OUTPUT"
-          fi
+          echo $TAG
+          echo $LABEL
+          # This allows us to parse the case when there are multiple tags. 
+          tag=$(echo $TAG | sed -e "s+[^:]*[:]\([^ ]*\).*+\1+")
+          echo "tag=$tag" >> "$GITHUB_OUTPUT"
+          echo "name=tag-$tag" >> "$GITHUB_OUTPUT"
```

Comment:
the metadata action generates tags like registry/image:tag. This extracts the tag from that.